packages/itwinui-react/src/core/Select/Select.tsx

```diff
@@ -267,13 +286,29 @@ const CustomSelect = React.forwardRef((props, forwardedRef) => {
     triggerProps,
     status,
     popoverProps,
+    defaultValue: defaultValueProp,
     ...rest
   } = props;
 
   const [isOpen, setIsOpen] = React.useState(false);
   const [liveRegionSelection, setLiveRegionSelection] = React.useState('');
 
-  const [uncontrolledValue, setUncontrolledValue] = React.useState<unknown>();
+  const [uncontrolledValue, setUncontrolledValue] = React.useState<unknown>(
+    (() => {
+      if (defaultValueProp !== undefined) {
+        return defaultValueProp;
+      }
+
+      // If borderless, empty value is not allowed. So, select the first option.
+      if (styleType === 'borderless') {
+        return isMultipleEnabled(valueProp, multiple)
+          ? [options[0]?.value]
+          : options[0]?.value;
+      }
+
+      return undefined;
+    })(),
+  );
```

Comment:
I believe we didn't previously support `defaultValue` in CustomSelect, so all this is not necessary for the purpose of this PR.

Also see [related thread](https://github.com/iTwin/iTwinUI/pull/1886#discussion_r1516798133)

Reply:
Yes, we didn't support it previously for `CustomSelect`. I thought that was a bug and so I fixed it here.

Are you suggesting that this fix can come in a separare PR? Or are you suggesting reverting this altogether?

Reply:
It should be reverted, at least for now. This PR's main purpose is: "add `styleType="borderless"` support to `<Select native>`".

_If_ we want to support `defaultValue`, it should happen in a separate PR.

Reply:
Sounds good, reverted (https://github.com/iTwin/iTwinUI/pull/1886/commits/6eabe377fd4b159eb66a99a84d8a78d1c66e6322). Added this comment to the after-PR TODOs.